tests/test_compute_service.py

```diff
@@ -408,3 +409,50 @@ def cleanup_router(router, gateway):
                             fip.in_use,
                             "Attached floating IP %s address should be in use."
                             % fip.public_ip)
+
+    @helpers.skipIfNoService(['compute.instances', 'networking.networks',
+                              'security.vm_firewalls'])
+    def test_instance_start_stop_methods(self):
+        label = "cb-instmethods-{0}".format(helpers.get_uuid())
+
+        # Declare these variables and late binding will allow
+        # the cleanup method access to the most current values
+        net = None
+        test_inst = None
+        fw = None
+        with cb_helpers.cleanup_action(lambda: helpers.cleanup_test_resources(
+                instance=test_inst, vm_firewall=fw, network=net)):
+            net = self.provider.networking.networks.create(
+                label=label, cidr_block=BaseNetwork.CB_DEFAULT_IPV4RANGE)
+            cidr = '10.0.1.0/24'
+            subnet = net.subnets.create(label=label, cidr_block=cidr)
+            test_inst = helpers.get_test_instance(self.provider, label,
+                                                  subnet=subnet)
```

Comment:
```suggestion
        label = "cb-instmethods-{0}".format(helpers.get_uuid())

        # Declare these variables and late binding will allow
        # the cleanup method access to the most current values
        subnet = None
        test_inst = None
        with cb_helpers.cleanup_action(lambda: helpers.cleanup_test_resources(
                instance=test_inst)):
            subnet = helpers.get_or_create_default_subnet(self.provider)
            test_inst = helpers.get_test_instance(self.provider, label,
                                                  subnet=subnet)
```

Reply:
hey @nuwang made the changes that you suggested (btw thanks for them). For now I updated unit tests so that it will conditonally skip for all provider but AWS, since I wasn't sure about other providers.